### PR TITLE
feat: safe drop down selector

### PR DESCRIPTION
**Pull request by ChillYao**
> Dropdown grows a trusted shelf,
> bookmarks pin, skeletons load,
> ENS names resolve, accounts unfold.

## What it solves

https://linear.app/safe-global/issue/WA-1828/enhance-safeselectordropdown-for-all-flows

Implements the Safe Selector Dropdown enhancements and All Accounts modal for the Spaces M2 milestone:

- Users had no way to manage trusted Safes directly from the selector dropdown
- Non-pinned Safes navigated via URL would not appear in the selector or chain switcher
- No copy-to-clipboard functionality on Safe addresses in the selector
- No loading/error states in the dropdown
- ENS names were not displayed anywhere in the selector or search results

### All Accounts Modal (new)
- Full "All Accounts" modal triggered by clicking "All Accounts >" button in the dropdown footer
- Shows **all** Safes (not just pinned) with single-chain and multi-chain grouping
- **Pin/unpin toggle**: filled bookmark = trusted, empty bookmark = not trusted. Unpin does not remove the card
- Multi-chain Safes show as flat cards (no collapsible expansion, no click navigation)
- Search by name or address
- Per-Safe: identicon, name (with ENS fallback), shortened address, copy button, chain logo(s), balance, threshold badge, Read-only / Not activated status badges
- Rename via three-dot context menu (uses MUI `EntryDialog` with elevated z-index)
- "Add existing" / "Create new" footer buttons
- Section header: "Manage trusted Safes" (plain text label)
- Loading skeleton while data is fetched

### Selector Dropdown enhancements
- **Dropdown header**: "Trusted Safes" label (plain text) + bookmark icon to pin/unpin the current Safe
- **Dropdown footer**: "All Accounts >" button opens the All Accounts modal
- **Copy address button** on the trigger content (works in both named and address-only states, blocks pointer-down to prevent dropdown toggle)
- **Loading skeleton** and **error state with retry** inside the dropdown content
- **`forceOpenable`**: dropdown can open even with a single Safe when header/footer are present, so users always have access to trusted Safe management
- Balance display changed from **all-chain total** to **current chain only**

### Safe injection for non-pinned Safes
- `useSafeBarSafes` now injects the current Safe into both `dropdownSafes` and `chainSelectorSafes` when it's not pinned or in a Space — constructs a fallback `SafeItem` from `useSafeInfo` + `useChainId` so the selector and chain switcher always render

### ENS name display
- `useSafeDisplayName` now uses `useAddressResolver` for ENS fallback. Priority: preferred name > address book > ENS > empty string
- ENS names display in: selector dropdown trigger, dropdown list items, All Accounts modal cards, and global search results (`SafeCardReadOnly`)
- ENS results are cached in-memory per chain+address by `useAddressResolver` (one lookup per address per session)

### z-index unification
- `dialog.tsx`: changed Dialog backdrop and content from `z-[9200]` to `z-[var(--z-overlay)]` (1400), so all shadcn overlay components (Dialog, DropdownMenu, Tooltip) share the same z-index layer. Radix's stacking context handles the layering automatically
- This fixes the DropdownMenu-inside-Dialog z-index issue (context menu was invisible behind Dialog backdrop)
- MUI `EntryDialog` uses `sx={{ zIndex: 1500 }}` to render above shadcn overlays

### Analytics

Events carried over from MyAccounts:

| Event | Trigger location |
|-------|-----------------|
| `PIN_SAFE` (pin/unpin) | Dropdown header bookmark, All Accounts modal bookmark toggle |
| `OPEN_SAFE` | Legacy Accounts modal (PinnedSafeItem): click Safe to navigate |
| `EXPAND_MULTI_SAFE` | Legacy Accounts modal (PinnedMultiSafeItem): expand multi-chain group |
| `SEARCH` | All Accounts modal search input (debounced, fires once per query) |
| `ADD_TO_WATCHLIST` | All Accounts modal "Add existing" button (label: `top_bar`) |
| `CREATE_NEW_SAFE` | All Accounts modal "Create new" button (label: `top_bar`) |

Events NOT added (not applicable in this context):

| Event | Reason |
|-------|--------|
| `SORT_SAFES` | No sort controls in this milestone |
| `TRUSTED_SAFES_ADDED/REMOVED` (batch) | Batch pin/unpin only in SafeSelectionModal (already tracked there) |
| `TRUSTED_SAFES_WARNING_SHOW` / `ADD_SINGLE` / `_CONFIRM` | Trust dialog not implemented here |
| `TRUSTED_SAFES_MIGRATION_PROMPT` | Migration prompt not implemented here |
| `TRUSTED_SAFES_SIMILAR_ADDRESS_CONFIRM` | Similarity detection not in this component |
| `TOTAL_SAFES_OWNED` / `TOTAL_SAFES_PINNED` | Meta events, only fire on login page |
| `EXPORT_DATA` / `IMPORT_DATA` | Data widget not in this component |

### Other
- `usePinActions` exported from `@/features/myAccounts` barrel (fixes eslint `no-restricted-imports`)
- `EntryDialog` accepts optional `sx` prop for z-index override
- Hover background color unified to `hover:bg-muted/30` across TopBar, modal cards, and dropdown items
- All new components are **shadcn/Tailwind only** (no MUI inside the All Accounts modal, except `EntryDialog` for rename)

## How to test it

1. **Dropdown header/footer**
   - Open the Safe selector dropdown → see "Trusted Safes" header with bookmark icon and "All Accounts >" footer button
   - Click "All Accounts >" → All Accounts modal opens

2. **All Accounts modal**
   - Shows all Safes (pinned and unpinned)
   - Search filters by name or address
   - Click empty bookmark → pin Safe (filled bookmark, notification). Click again → unpin (empty bookmark, notification). Card stays in list
   - Multi-chain Safes show as flat cards, no expand/collapse
   - Click three-dot menu → "Rename" or "Give name" → MUI rename dialog opens above the modal
   - "Add existing" / "Create new" navigate to respective pages

3. **Pin/unpin from dropdown header**
   - Navigate to a non-pinned Safe → dropdown header shows empty bookmark
   - Click bookmark → pinned (filled). Click again → unpinned

4. **Non-pinned Safe visibility**
   - Navigate directly to a Safe URL that isn't pinned → selector dropdown and chain switcher still render
   - Only one Safe, no others pinned → dropdown still opens (can access header/footer)

5. **Copy address**
   - In selector trigger: click copy icon next to address → copies to clipboard, icon changes to check
   - In All Accounts modal: copy icon next to each Safe's address
   - Copy does NOT toggle the dropdown open/closed

6. **Loading/error states**
   - Throttle network → All Accounts modal shows skeleton rows; dropdown shows skeleton items
   - Block API → dropdown shows error state with "Reload" button

7. **ENS names**
   - Navigate to a Safe that has an ENS name → name displays in selector trigger, dropdown items, All Accounts modal cards, and global search results

## Visual summary

```mermaid
flowchart TB
    subgraph SafeSelectorDropdown
        Trigger["Trigger: identicon + name/ENS + address + copy + balance"]
        Header["Header: Trusted Safes | 🔖 pin/unpin"]
        Items[SelectItems: other Safes in list]
        Footer["Footer: All Accounts >"]
    end

    subgraph AllAccountsModal
        Search[Search bar]
        List["All Safe cards with pin/unpin toggle"]
        Buttons["Add existing | Create new"]
    end

    subgraph useSafeBarSafes
        Pinned[pinnedSafes]
        AllKnown[allKnownSafes]
        Fallback[fallbackCurrentSafe]
        Inject{Current Safe in list?}
        Inject -->|No| Fallback --> DropdownSafes
        Inject -->|Yes| DropdownSafes[dropdownSafes]
    end

    subgraph useSafeDisplayName
        AddrBook[Address book name]
        ENS[ENS name via useAddressResolver]
        Priority["preferredName > addrBook > ENS"]
    end

    Footer -->|click| AllAccountsModal
    DropdownSafes --> Items
    useSafeDisplayName --> Trigger
    useSafeDisplayName --> List
```

## Checklist

- [ ] I've tested the branch on mobile 📱
- [x] I've documented how it affects the analytics (if at all) 📊
- [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

### CODE REVIEW

**Comment by ChillYao** [comment]
@claude review

**Comment by ChillYao** [comment]
> **Claude finished @ChillYao's task** —— [View job](https://github.com/safe-global/safe-wallet-monorepo/actions/runs/24498959760)
> 
> ### PR Code Review
> #### Todo List
> * [x]  Review core hook implementations
> * [x]  Check new component architecture
> * [x]  Analyze test coverage and quality
> * [x]  Verify adherence to repo guidelines
> * [x]  Check for potential issues and improvements
> 
> ### Overall Assessment
> This is a well-implemented feature with solid architecture and comprehensive test coverage. The code follows repository conventions and demonstrates good separation of concerns. Here are the key findings:
> 
> ### ✅ Strengths
> 1. **Excellent Hook Design**
>    
>    * `useSafeBarSafes` cleanly handles Safe injection logic with proper fallback patterns
>    * `useSafeSelectorState` manages complex dropdown state elegantly with `forceOpenable` logic
>    * Both hooks have comprehensive test coverage (95%+ statement coverage)
> 2. **Component Architecture**
>    
>    * Proper separation between container (`SafeDropdownContainer`) and presentational components
>    * Consistent use of shadcn/Tailwind throughout (no MUI mixing)
>    * Loading/error states properly implemented with skeleton components
> 3. **Analytics Integration**
>    
>    * Comprehensive analytics tracking matching the existing MyAccounts component
>    * Proper event debouncing for search (300ms delay)
>    * Clear tracking of user interactions
> 4. **Storybook Coverage**
>    
>    * Excellent story coverage including edge cases (loading, error, single safe)
>    * Interactive wrapper for state management demonstrations
>    * Good use of mock data generation
> 
> ### ⚠️ Areas for Improvement
> #### 1. Accessibility Issues
> ```ts
> // AccountsModal/PinnedSafeItem.tsx:152
> <button
>   type="button"
>   onClick={handleUnpin}
>   className="shrink-0 rounded p-1 hover:bg-muted"
>    // ✅ Good
> >
> ```
> 
> **Missing:** Keyboard navigation support and focus management in the AccountsModal. Consider adding:
> 
> * Tab order management
> * Arrow key navigation for the safe list
> * Escape key handling for modal closure
> 
> #### 2. Performance Considerations
> ```ts
> // AccountsModal/index.tsx:32
> const allItems = useMemo<AllSafeItems>(() => {
>   const multi = allMultiChainSafes ?? []
>   const single = allSingleSafes ?? []
>   return [...multi, ...single].sort((a, b) => (b.lastVisited ?? 0) - (a.lastVisited ?? 0))
> }, [allMultiChainSafes, allSingleSafes])
> ```
> 
> **Issue:** Sorting happens on every render. For large numbers of safes, consider:
> 
> * Moving sort logic to the data layer
> * Using `useMemo` with more granular dependencies
> * Implementing virtualization for 100+ safes
> 
> #### 3. Type Safety Enhancement
> ```ts
> // types.ts:54
> export interface SafeItemData {
>   id: string
>   name?: string | null
>   address: string
>   // ...
>   parentSafeId?: string  // ⚠️ No validation that this ID exists
> }
> ```
> 
> **Suggestion:** Add runtime validation or use branded types to ensure `parentSafeId` references valid safes.
> 
> #### 4. Error Handling
> ```ts
> // SafeDropdownContainer.tsx:67-68
> if (isError) {
>   return <DropdownContentError onRetry={onRetry} />
> }
> ```
> 
> **Missing:** Specific error types and user-friendly error messages. Consider different handling for:
> 
> * Network errors vs. data parsing errors
> * Rate limiting vs. server errors
> * Provide more specific retry guidance
> 
> #### 5. Search UX Enhancement
> ```ts
> // AccountsModal/index.tsx:48-58
> useEffect(() => {
>   // Search tracking with 300ms debounce
> }, [search])
> ```
> 
> **Enhancement opportunities:**
> 
> * Add search highlighting in results
> * Show "no results" state with search suggestions
> * Consider fuzzy search for better UX (addresses are hard to type exactly)
> 
> ### 🔧 Minor Issues
> 1. **Magic Numbers:** `SKELETON_COUNT = 4` could be configurable or responsive to viewport size
> 2. **Hardcoded Styles:** Some pixel values could use design tokens (e.g., `size-4`, `gap-3`)
> 3. **Bundle Size:** Consider lazy loading the AccountsModal since it's a heavy component
> 
> ### 📊 Test Quality
> * **Coverage:** Excellent (95%+ on new files)
> * **Test Structure:** Well-organized with clear test helpers
> * **Edge Cases:** Good coverage of fallback scenarios and error states
> * **Mocking:** Proper mocking patterns following repository conventions
> 
> ### 🎯 Repository Adherence
> * ✅ Follows feature architecture patterns (proper barrel exports)
> * ✅ Uses shadcn/Tailwind exclusively (no MUI in new components)
> * ✅ Proper analytics integration with existing event structure
> * ✅ Test files co-located with source files
> * ✅ Storybook stories for all major components
> 
> ### 🚀 Recommendation
> **APPROVE** - This is a high-quality implementation that enhances the user experience significantly. The identified issues are minor and can be addressed in follow-up PRs if needed.

**1. Accessibility (keyboard nav):** Agreed this would be nice, but it's scoped as "Nice to have" in the AC. The shadcn Dialog already handles Escape and Tab focus trapping out of the box. Arrow key list navigation is a follow-up.

**2. Performance (sorting):** The sort is already inside `useMemo` with `[allMultiChainSafes, allSingleSafes]` as deps — it only re-runs when the underlying data changes, not on every render. Pinned safes are typically <20, so virtualization would be overkill here.

**3. Type Safety (parentSafeId):** This type is pre-existing in `SafeItemData`, not something we introduced. Changing it would be a separate refactor.

**4. Error Handling (specific error types):** Fair point in general, but from the user's perspective there's not much they can do differently for a network error vs a parsing error — retry is the action either way. Keeping it simple for now.

**5. Search UX (highlighting, fuzzy):** Good ideas for a follow-up. Current exact match works fine for addresses and names. Fuzzy search would need a dependency (like fuse.js) which adds bundle weight — worth a separate discussion.

**Minor issues:**
- `SKELETON_COUNT = 4` works across viewports, no real benefit making it dynamic
- `size-4`, `gap-3` etc. are Tailwind spacing tokens, not hardcoded pixel values
- AccountsModal already returns `null` when closed so it doesn't render anything — lazy loading on top of that wouldn't buy much

**Comment by ChillYao** [comment]
@claude review

**Comment by ChillYao** [comment]
> **Claude finished @ChillYao's task** —— [View job](https://github.com/safe-global/safe-wallet-monorepo/actions/runs/24513481583)
> 
> ### PR Code Review
> #### Todo List
> * [x]  Review core implementation files
> * [x]  Check for specific issues and improvements
> * [x]  Provide focused feedback on code quality
> * [x]  Update with final review
> 
> After reviewing the implementation, I found several specific areas for improvement beyond the previous review feedback:
> 
> ### 🔧 Concrete Issues to Address
> #### 1. **Potential Memory Leak in Search Debouncing**
> ```ts
> // AccountsModal/index.tsx:53-57
> const timer = setTimeout(() => {
>   trackEvent(OVERVIEW_EVENTS.SEARCH)
>   searchTracked.current = true
> }, 300)
> return () => clearTimeout(timer)
> ```
> 
> **Issue**: The cleanup function only clears the timer, but `searchTracked.current` remains `true` and never resets when the component unmounts or search changes quickly.
> 
> **Fix**: Reset the flag in cleanup:
> 
> ```ts
> return () => {
>   clearTimeout(timer)
>   searchTracked.current = false
> }
> ```
> 
> #### 2. **Inefficient Array Operations**
> ```ts
> // useSafeBarSafes.ts:30-32, 35-37
> const pinnedSafes = useMemo<AllSafeItems>(
>   () => [...(pinnedGrouped.allMultiChainSafes ?? []), ...(pinnedGrouped.allSingleSafes ?? [])],
>   [pinnedGrouped.allMultiChainSafes, pinnedGrouped.allSingleSafes],
> )
> ```
> 
> **Issue**: Double spread operation creates unnecessary intermediate arrays. This pattern repeats multiple times.
> 
> **Fix**: Use `concat()` for better performance:
> 
> ```ts
> const pinnedSafes = useMemo<AllSafeItems>(
>   () => (pinnedGrouped.allMultiChainSafes ?? []).concat(pinnedGrouped.allSingleSafes ?? []),
>   [pinnedGrouped.allMultiChainSafes, pinnedGrouped.allSingleSafes],
> )
> ```
> 
> #### 3. **Missing Error Boundary for Safe Injection Logic**
> ```ts
> // useSafeBarSafes.ts:58-60
> const current = allKnownSafes.find((s) => sameAddress(s.address, safeAddress)) ?? fallbackCurrentSafe
> if (!current) return pinnedSafes
> return [current, ...pinnedSafes]
> ```
> 
> **Issue**: If `fallbackCurrentSafe` creation fails silently (e.g., invalid chainId), the dropdown may not render the current safe without any indication to the user.
> 
> **Improvement**: Add error handling or at minimum console warning for debugging.
> 
> #### 4. **Type Safety Gap in Modal State**
> ```ts
> // AccountsModal/index.tsx:63
> <Dialog open onOpenChange={(isOpen) => !isOpen && onClose()}>
> ```
> 
> **Issue**: The Dialog always has `open={true}` but uses `onOpenChange` for closing. This creates inconsistent state management - the Dialog thinks it's controlled but doesn't receive the actual open state.
> 
> **Fix**: Pass the actual `open` prop:
> 
> ```ts
> <Dialog open={open} onOpenChange={(isOpen) => !isOpen && onClose()}>
> ```
> 
> #### 5. **Hardcoded Magic Numbers**
> ```ts
> // SafeDropdownContainer.tsx:89
> className="w-[430px] max-w-[calc(100vw-2rem)] max-h-[20rem]"
> ```
> 
> **Issue**: Hardcoded pixel values make responsive design harder. Consider using CSS custom properties or Tailwind design tokens.
> 
> #### 6. **Potential Race Condition**
> ```ts
> // useSafeSelectorState.ts:32-35  
> useEffect(() => {
>   const chainId = getFirstChainId(selectedItem)
>   setSelectedChainId(chainId)
> }, [selectedItem])
> ```
> 
> **Issue**: If `selectedItem` changes rapidly, there could be stale closures. The effect should handle cleanup.
> 
> ### 💡 Performance Optimizations
> 1. **Memoize expensive operations**: The address filtering in `AccountsModal` could benefit from `useCallback` for the filter function
> 2. **Consider virtualization**: If the safe list grows large (>50 items), implement windowing
> 3. **Optimize re-renders**: Use `React.memo` for `SafeItemCard` since it receives complex props
> 
> ### ✅ What's Done Well
> * Comprehensive test coverage with realistic scenarios
> * Proper separation of concerns between hooks and components
> * Consistent error handling patterns
> * Good TypeScript typing throughout
> * Clean component composition with proper prop drilling
> 
> The implementation is solid overall - these issues are mostly about edge cases and performance optimization rather than fundamental problems.

**1. Search debounce "leak":** Not a leak — `searchTracked` is a ref that gets GC'd on unmount. It already resets when search is cleared (line 49-51). The suggested fix would actually cause duplicate tracking on fast typing.

**2. Spread vs concat:** Negligible difference for <50 items. `[...a, ...b]` is the project's convention — consistency over micro-optimization.

**3. Error boundary for safe injection:** Already handled — `fallbackCurrentSafe` returns `undefined` when inputs are invalid, and the next line (`if (!current) return pinnedSafes`) falls back gracefully. No silent failure.

**4. Dialog open state:** Intentional pattern. The component does `if (!open) return null` before rendering, so `<Dialog open>` is always true when it exists. This avoids rendering content during close animations. Same pattern used elsewhere in the project.

**5. Hardcoded 430px:** Pre-existing value, not introduced by this PR. Already responsive via `max-w-[calc(100vw-2rem)]`.

**6. Race condition:** Pre-existing code, not our change. No actual race — `setSelectedChainId` is synchronous and the effect re-runs on every `selectedItem` change. No async operations involved.

**Performance suggestions:** Filter is already in `useMemo`, virtualization is overkill for <50 items, and `React.memo` on `SafeItemCard` would add overhead since props change on every pin/unpin.

File: apps/web/src/hooks/useSafeDisplayName.ts [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
- *
- * The proper fix is to cache ENS results (including null) in the Redux store so each
- * address is only resolved once per session. Until that is implemented, ENS is omitted
- * to avoid excessive RPC calls that risk hitting rate limits (HTTP 429).
+ * ENS results are cached in-memory per chain+address by useAddressResolver,
+ * so each address is only resolved once per session.
  */
 export const useSafeDisplayName = (address: string, chainId: string, preferredName?: string): string => {
   const addressBookItem = useAddressBookItem(address, chainId)
+  const { ens } = useAddressResolver(address)
```
Comment (vmosharova) [inline apps/web/src/hooks/useSafeDisplayName.ts:13]: The comment was deleted:
> The proper fix is to cache ENS results (including null) in the Redux store so each

Do we cache null ENS results? If no, then we risk 429s because if the safe has no ENS name, then we will have an RPC call on every mount for addresses without ENS name.

File: apps/web/src/components/common/SpaceSafeBar/hooks/__tests__/useSafeBarSafes.test.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
+import { renderHook } from '@testing-library/react'
```
Comment (vmosharova) [inline apps/web/src/components/common/SpaceSafeBar/hooks/__tests__/useSafeBarSafes.test.ts:1]: Let's move all hooks test files to a __tests__ subfolder for better code structure.

File: apps/web/src/components/common/SpaceSafeBar/hooks/useSpaceSafeSelectorItems.ts [hunk 3]
```diff
@@ -63,19 +58,17 @@ function buildMultiChainItem(
...
 
   return {
     id: `${orderedChainIds[0]}:${item.address}`,
     name: item.name ?? '',
     address: item.address,
-    ...resolveThresholdAndOwners(isCurrentSafe, safe, firstOverview),
-    balance: sumFiatTotals(safeOverviews).toString(),
-    isLoading: overviewsLoading && safeOverviews.length === 0,
+    ...resolveThresholdAndOwners(isCurrentSafe, safe, currentChainOverview),
+    balance: currentChainOverview?.fiatTotal ?? '0',
```
Comment (vmosharova) [inline apps/web/src/components/common/SpaceSafeBar/hooks/useSpaceSafeSelectorItems.ts:70]: Seems like the balance was changes from the sum across all chains to the current chain - it is a product requirement?

File: apps/web/src/components/common/SpaceSafeBar/hooks/useSafeBarSafes.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  const pinnedSafes = useMemo<AllSafeItems>(
+    () => [...(pinnedGrouped.allMultiChainSafes ?? []), ...(pinnedGrouped.allSingleSafes ?? [])],
+    [pinnedGrouped.allMultiChainSafes, pinnedGrouped.allSingleSafes],
+  )
+
+  const allKnownSafes = useMemo<AllSafeItems>(
+    () => [...(allGrouped.allMultiChainSafes ?? []), ...(allGrouped.allSingleSafes ?? [])],
+    [allGrouped.allMultiChainSafes, allGrouped.allSingleSafes],
+  )
```
Comment (vmosharova) [inline apps/web/src/components/common/SpaceSafeBar/hooks/useSafeBarSafes.ts]: A nit for a better readability:
```
const allKnownSafes = useMemo<AllSafeItems>(() => {
  const multiChainSafes = allGrouped.allMultiChainSafes ?? []
  const singleSafes = allGrouped.allSingleSafes ?? []

  return [...multiChainSafes, ...singleSafes]
}, [allGrouped.allMultiChainSafes, allGrouped.allSingleSafes])
```

File: apps/web/src/components/common/SpaceSafeBar/AccountsModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            <span className="text-xs font-semibold uppercase tracking-wider text-muted-foreground">
+              Manage trusted Safes
+            </span>
+          </div>
+
+          {!allSafes ? (
+            <SafeListSkeleton />
+          ) : filteredItems.length === 0 ? (
+            <p className="px-2 py-6 text-center text-sm text-muted-foreground">
+              {search.trim() ? 'No safes match your search' : 'No safes yet'}
```
Comment (vmosharova) [inline apps/web/src/components/common/SpaceSafeBar/AccountsModal/index.tsx:124]: Does it make sense to show an error when there was some error (e.g. in backend) when fetching Safes?

File: apps/web/src/components/sidebar/NestedSafesPopover/index.tsx [hunk 6]
```diff
@@ -311,7 +311,11 @@ export function NestedSafesPopover({
       open={!!anchorEl}
       anchorEl={centered ? undefined : anchorEl}
       anchorReference={centered ? 'anchorPosition' : 'anchorEl'}
-      anchorPosition={centered ? { top: window.innerHeight / 2, left: window.innerWidth / 2 } : undefined}
+      anchorPosition={
+        centered && typeof window !== 'undefined'
```
Comment (vmosharova) [inline apps/web/src/components/sidebar/NestedSafesPopover/index.tsx]: Might be a better approach is to check if the component mounted instead of checking the `window`

**Review by vmosharova — CHANGES_REQUESTED** [review]
Great job! Left some comments :)

File: apps/web/src/hooks/useSafeDisplayName.ts (hunk 1 above)
Comment (ChillYao) [reply apps/web/src/hooks/useSafeDisplayName.ts:13]: No. I decided not change anything about `useAddressResolver`. It is too risky.
If this is an important issue, maybe we should open up another ticket, in my view.

File: apps/web/src/components/common/SpaceSafeBar/hooks/useSpaceSafeSelectorItems.ts (hunk 3 above)
Comment (ChillYao) [reply apps/web/src/components/common/SpaceSafeBar/hooks/useSpaceSafeSelectorItems.ts:70]: Yes. @devpramoth told me that in person. I sent a message in channel to double check with him.

File: apps/web/src/components/sidebar/NestedSafesPopover/index.tsx (hunk 6 above)
Comment (ChillYao) [reply apps/web/src/components/sidebar/NestedSafesPopover/index.tsx]: Very good catch!
Claude suggest using `anchorEl` as the guard instead of a separate mounted state — anchorEl is null during SSR by definition, and it's also the condition for open, so it naturally expresses "we only need this when the popover is actually anchored".

**Review by vmosharova — APPROVED** [review]
👍🏻
